### PR TITLE
protect _text member of Text3DOverlay with mutex

### PR DESCRIPTION
**Pull request by davidkelly**
This fixes a crash bug reported in https://highfidelity.fogbugz.com/f/cases/4873/CRASH-without-Bugsplat-standing-around-listening-to-people-talking.   We need to protect the text member of this class, and not pass around references to it, as you can catch it in the middle of updating.  QString does some internal allocation, and trying to use it in the middle of that will cause crashes.

### CODE REVIEW

**Comment by davidkelly** [comment]
Test Plan
I have a [simple script](https://github.com/highfidelity/hifi/files/1049058/overlayTest.txt) which can reproduce the crash.  Run it on master, you should see it crash eventually.  For me, I saw it in a few minutes.  Then, run it with this PR.  I ran for 30 min, no issues.

**Comment by davidkelly** [comment]
FWIW I'd guess there are similar QString issues lurking out there with the overlays -- I'll take a look and file a bug if needed...

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/385/HighFidelity-Beta-PR10611-f9e1f028882492fc5a5e9ccc17416660e69cdfb4.dmg

File: interface/src/ui/overlays/Text3DOverlay.cpp [hunk 1]
```diff
@@ -23,14 +23,15 @@ const float LINE_SCALE_RATIO = 1.2f;
 
 QString const Text3DOverlay::TYPE = "text3d";
 
-Text3DOverlay::Text3DOverlay() {
+Text3DOverlay::Text3DOverlay() : _mutex() {
```
Comment (jherico) [inline interface/src/ui/overlays/Text3DOverlay.cpp]: you don't need to add this initializer

File: interface/src/ui/overlays/Text3DOverlay.cpp [hunk 2]
```diff
@@ -23,14 +23,15 @@ const float LINE_SCALE_RATIO = 1.2f;
...
     _textRenderer = TextRenderer3D::getInstance(SANS_FONT_FAMILY, FIXED_FONT_POINT_SIZE);
     _geometryId = DependencyManager::get<GeometryCache>()->allocateID();
 }
 
 Text3DOverlay::Text3DOverlay(const Text3DOverlay* text3DOverlay) :
     Billboard3DOverlay(text3DOverlay),
     _text(text3DOverlay->_text),
+    _mutex(),
```
Comment (jherico) [inline interface/src/ui/overlays/Text3DOverlay.cpp]: don't need to add an explicit initializer

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/390/HighFidelity-Beta-PR10611-12cbc206d2fc94f87e0df6916f27cf81df6b0a0e.exe

**Comment by davidkelly** [comment]
test this please

**Comment by YuppyPlays** [comment]
Okay to QA when built?

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/408/HighFidelity-Beta-PR10611-12cbc206d2fc94f87e0df6916f27cf81df6b0a0e.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/408/HighFidelity-Beta-PR10611-12cbc206d2fc94f87e0df6916f27cf81df6b0a0e.dmg

**Comment by davidkelly** [comment]
Yes -- ready to QA

**Comment by YuppyPlays** [comment]
Testing.

**Comment by YuppyPlays** [comment]
Testing complete:

So, I tested this in the current master (6644) and I didn't experience a crash weirdly. Just had slightly more cpu usage until i ended it (ran for 20 minutes)

Same exact results for the PR build. (20 minutes)

Unsure if this is a **Pass?**

**Comment by davidkelly** [comment]
Maybe I was lucky to trigger the crash, even with this script.  I just sat there and watched the random numbers in the overlay until after awhile - 5 min?  I'll rerun a few times, but I think in any case this is ok and we can merge.  Will ponder tonight

**Comment by davidkelly** [comment]
I was able to repro it with a build off master again with this script.  I believe it may be a matter of luck.  Calling it good.

**Pull request merged**